### DISCUSSION TITLE
Ability to export a collection without records

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#1413] [Q&A]
Is there a way to export a collection without taking the records?  It would be nice for quickly and crudely saving the schema without needing to manually modify the output.

**Comment by ganigeorgiev** [comment]
I don't understand what do you mean by "_export a collection without taking the records_".

Do you mean you want to fetch a `models.Collection` instance or just its schema?
You can fetch a collection by following some of the examples from https://pocketbase.io/docs/collection-methods/. You can access the collection schema fields by `collection.Schema.Fields()`.

Or do you mean something completely else?

**Reply by ganigeorgiev** [reply]
Additionally, you can export all your collections configuration from the "Admin UI > Settings > Export collections".

You can also create a migration snapshot with all of your collection configurations via `migrate collections`.

**Reply by dschissler** [reply]
> I don't understand what do you mean by "_export a collection without taking the records_".
> 
> Do you mean you want to fetch a `models.Collection` instance or just its schema? You can fetch a collection by following some of the examples from https://pocketbase.io/docs/collection-methods/. You can access the collection schema fields by `collection.Schema.Fields()`.
> 
> Or do you mean something completely else?

I meant about exporting the schema only.

**Reply by ganigeorgiev** [reply]
You can use the JSON from the "Admin UI > Settings > Export collections".

If you want to copy/transfer your collections to another environment you can import the previous json in "Admin UI > Settings > Import collections" or use the migration tool - https://pocketbase.io/docs/migrations/.

PocketBase v0.9+ comes also with automigrations support so any collection change from the Admin UI should result in a migration file inside `pb_migrations`. All generated files will be automatically picked on `serve` when deployed in a new environment.
